data/provider/src/main/java/tech/pegasys/teku/api/stateselector/StateSelectorFactory.java

```diff
@@ -79,20 +85,45 @@ public StateSelector genesisSelector() {
 
   @Override
   public StateSelector finalizedSelector() {
+
     return () ->
         SafeFuture.completedFuture(
             client
                 .getLatestFinalized()
                 .map(
-                    finalized ->
-                        addMetaData(
-                            finalized.getState(),
-                            // The finalized checkpoint may change because of optimistically
-                            // imported blocks at the head and if the head isn't optimistic, the
-                            // finalized block can't be optimistic.
-                            client.isChainHeadOptimistic(),
-                            true,
-                            true)));
+                    finalized -> {
+                      final UInt64 slot = finalized.getSlot();
+                      final Spec spec = finalized.getSpec();
```

Comment:
we have spec already at `StateSelectorFactory`

Reply:
how did I miss that?